### PR TITLE
SCIM Update error code and message

### PR DESCRIPTION
**Pull request by aditya-ven**
[MRN-408](https://vendasta.jira.com/browse/MRN-408)
@vendasta/marina 
@vendasta/external-apis

**TODO**:
- [x] Review the [pre release checklist](https://vendasta.jira.com/wiki/spaces/API/pages/1651769533/Pre+Release+Checks)

[MRN-408]: https://vendasta.jira.com/browse/MRN-408?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: openapi/scim/scim.yaml [hunk 1]
```diff
@@ -95,8 +95,11 @@ paths:
             application/scim+json:
               schema:
                 $ref: '#/components/schemas/ListResponseUsers'
-        '404':
-          description: User not found
+        '400':
+          description: |-
+            Invalid input
+            <br>Invalid Filter
```
Comment (malamsk) [inline openapi/scim/scim.yaml:101]: It should have the exact error message the api is going to provide.
Like for example if we provide filter other than userName or externalId  the error message will be `Only "userName" "externalId" filter attribute supported.`

File: openapi/scim/scim.yaml [hunk 2]
```diff
@@ -122,8 +125,15 @@ paths:
             application/scim+json:
               schema:
                 $ref: '#/components/schemas/User'
-        '405':
-          description: Invalid input
+        '400':
+          description: Failed to create user.
+        '409':
+          description: One or more of the attribute values are already in use or are reserved.
+          content:
```
Comment (malamsk) [inline openapi/scim/scim.yaml:132]: 409 will not have any content

File: openapi/scim/scim.yaml (hunk 1 above)
Comment (aditya-ven) [reply openapi/scim/scim.yaml]: Changes are made.

File: openapi/scim/scim.yaml (hunk 2 above)
Comment (aditya-ven) [reply openapi/scim/scim.yaml]: Done.

**Review by malamsk — APPROVED** [review]

**Pull request merged**